chcemvediet/apps/accounts/tests/test_backend.py

```diff
@@ -0,0 +1,100 @@
+import mock
+
+from django.conf.urls import patterns, url
+from django.contrib.auth import get_user
+from django.contrib.auth.models import User
+from django.http import HttpResponse
+from django.test import TestCase
+from django.test.utils import override_settings
+
+from poleno.utils.http import _local
+
+
+class AdminLoginAsBackendTest(TestCase):
+
+    def mock_view(request):
+        return HttpResponse()
+
+    def set_admin_login_as_attribute_view(request, id):
+        request.session[u'admin_login_as'] = id
+        return HttpResponse()
+
+    urls = tuple(patterns(u'',
+        url(r'^$', mock_view),
+        url(r'^(.+)/login-as/$', set_admin_login_as_attribute_view),
+    ))
+
+    def create_users(self):
+        self.user = User.objects.create_user(
+                username=u'user',
+                email=u'user@example.com',
+                password=u'test',
+                )
+        self.superuser = User.objects.create_superuser(
+                username=u'superuser',
+                email=u'superuser@example.com',
+                password=u'test',
+                )
+
+    def setUp(self):
+        self.settings_override = override_settings(
+            AUTHENTICATION_BACKENDS=(u'chcemvediet.apps.accounts.backends.AdminLoginAsBackend',),
+            PASSWORD_HASHERS=(u'django.contrib.auth.hashers.MD5PasswordHasher',),
+        )
+        self.settings_override.enable()
+        self.create_users()
+
+    def tearDown(self):
+        self.settings_override.disable()
+
+
+    def test_get_user_returns_anonymous_user(self):
+        response = self.client.get(u'/')
+        request = response.wsgi_request
+        user = get_user(request)
+        self.assertIsNotNone(user)
+        self.assertTrue(user.is_anonymous())
+
+    def test_get_user_returns_logged_non_admin_user(self):
+        self.assertTrue(self.client.login(
+                username=self.user.username, password=u'test'
+        ))
+        response = self.client.get(u'/')
+        request = response.wsgi_request
+        _local.request = request
```

Comment:
@martinmacko47 Nevidel som to takto nikde pouzivat, ale ak som to nenastavil mi metoda `get_request` vratila `None` a cele to padlo.

Reply:
:ok: teraz som vlastne nasiel ze request sa ulozi do wsgi_request
```
# Attach the originating request to the response so that it could be
        # later retrieved.
        response.wsgi_request = request
```

Reply:
Zistil som preco sa to takto spravalo. V `AuthenticationMiddleware` pri spracovanie requestu sa vykona 
```
request.user = SimpleLazyObject(lambda: get_user(request))
```
. V mojom `mocked_view` sa to nikde nepouzilo az po skonceni endpointu pri assertovani na konci testu. `get_request` vratilo None, lebo `RequestProviderMiddleware` v `process_response` sa odstranil request: `_local.request = None`.